### PR TITLE
Adding homogeneous reconstruction protocol

### PR DESCRIPTION
**Pull request by fonsecareyna82**
* Adding homogeneous reconstruction protocol
*  Fixing an error continuing a protocol

### CODE REVIEW

File: cryosparc2/protocols/protocol_cryorefine.py [hunk 1]
```diff
@@ -437,13 +433,13 @@ def doRunRefine(self):
         """
         :return:
         """
-        if self.mask is not None:
-            input_group_connect = {"particles": str(self.par),
-                                   "volume": str(self.vol),
-                                   "mask": str(self.mask)}
+        if self.mask.get() is not None:
```
Comment (delarosatrevin) [inline cryosparc2/protocols/protocol_cryorefine.py:436]: This block of code seems repeated many times. Is there a way to refactor this to a common function or similar?

File: cryosparc2/protocols/protocol_cryorefine.py [hunk 2]
```diff
@@ -67,7 +67,7 @@ def _defineParams(self, form):
                       label="Input particles", important=True,
                       validators=[Positive],
                       help='Select the input images from the project.')
-        form.addParam('referenceVolume', PointerParam, pointerClass='Volume',
+        form.addParam('refVolume', PointerParam, pointerClass='Volume',
```
Comment (pconesa) [inline cryosparc2/protocols/protocol_cryorefine.py:70]: Renaming a params will break existing data compatibility. So, existing workflows (json) or saved data will not work.

File: cryosparc2/protocols/protocol_cryorefine.py [hunk 3]
```diff
@@ -285,13 +285,12 @@ def _insertAllSteps(self):
         self._defineFileNames()
         self._defineParamsName()
         self._initializeCryosparcProject()
-        self._insertFunctionStep("convertInputStep")
-        self._insertFunctionStep('processStep')
-        self._insertFunctionStep('createOutputStep')
+        self._insertFunctionStep(self.convertInputStep)
```
Comment (pconesa) [inline cryosparc2/protocols/protocol_cryorefine.py:288]: nice!

File: cryosparc2/protocols/protocol_cryosparc_helical_refinement.py [hunk 4]
```diff
@@ -64,7 +65,7 @@ def _defineParams(self, form):
                       validators=[Positive],
                       help='Particle stacks to use. Multiple stacks will '
                            'be concatenated.')
-        form.addParam('referenceVolume', PointerParam, pointerClass='Volume',
+        form.addParam('refVolume', PointerParam, pointerClass='Volume',
```
Comment (pconesa) [inline cryosparc2/protocols/protocol_cryosparc_helical_refinement.py:68]: same here renaming effect

File: cryosparc2/protocols/protocol_cryorefine.py (hunk 2 above)
Comment (delarosatrevin) [reply cryosparc2/protocols/protocol_cryorefine.py:70]: I agree, unless a very good reason I think it is better to keep existing param names in general

**Review by pconesa — COMMENTED** [review]
Cool refactoring. Only thing is the renaming of the referenceVolume to refVolume.

**Review by pconesa — APPROVED** [review]

**Pull request merged**